src/TLuaInterpreter.cpp

```diff
@@ -2614,42 +2558,37 @@ int TLuaInterpreter::setConsoleBufferSize(lua_State* L)
 {
     int s = 1;
     int n = lua_gettop(L);
-    std::string a1;
+    QString windowName;
     if (n > 2) {
         if (!lua_isstring(L, s)) {
             lua_pushstring(L, "setConsoleBufferSize: wrong argument type");
             lua_error(L);
             return 1;
-        } else {
-            a1 = lua_tostring(L, s);
-            s++;
         }
+        windowName = lua_tostring(L, s);
+        s++;
     }
-    int luaFrom;
+
     if (!lua_isnumber(L, s)) {
         lua_pushstring(L, "setConsoleBufferSize: wrong argument type");
         lua_error(L);
         return 1;
-    } else {
-        luaFrom = lua_tointeger(L, s);
-        s++;
     }
+    int luaFrom = lua_tointeger(L, s);
+    s++;
 
-    int luaTo;
     if (!lua_isnumber(L, s)) {
         lua_pushstring(L, "setConsoleBufferSize: wrong argument type");
         lua_error(L);
         return 1;
-    } else {
-        luaTo = lua_tointeger(L, s);
     }
+    int luaTo = lua_tointeger(L, s);
 
     Host& host = getHostFromLua(L);
 
-    if (a1 == "main" || n < 3) {
+    if (windowName.isEmpty() || windowName == "main" || n < 3) {
```

Comment:
Here you keep the `n < 3` clause. Was this by accident or am I missing some other difference?

Reply:
No, I missed that one.
Anyway it seems that I also need to go through with another fine-toothed comb and fix all the `empty name == missing name == name='main'` cases so it's remotely consistent. Will be in the next update.